docs/source/protocols.rst

```diff
@@ -409,3 +409,53 @@ in ``typing`` such as ``Iterable``.
    ``isinstance()`` with protocols is not completely safe at runtime.
    For example, signatures of methods are not checked. The runtime
    implementation only checks that all protocol members are defined.
+
+.. _callback_protocols:
+
+Callback protocols
+******************
+
+Protocols can be used to define flexible callback types that are impossible
```

Comment:
Nit: they are currently possible with the legacy extended callable syntax.

Reply:
Some of them are really impossible, like overloaded callbacks. I will try to clarify here.